pkg/model/api/v1/dashboard/variable.go

```diff
@@ -79,6 +79,23 @@ type VariableDisplay struct {
 	Hidden         bool `json:"hidden" yaml:"hidden"`
 }
 
+// UnmarshalJSON overrides the default json unmarshal behavior for the VariableDisplay struct
+// this fixes an issue with the Hidden attribute that gets ignored by the default unmarshal
+func (v *VariableDisplay) UnmarshalJSON(data []byte) error {
```

Comment:
I think you should just copy and past how it is done in `common.Display`.

So that would give something like that:

```go
func (d *VariableDisplay) UnmarshalJSON(data []byte) error {
	var tmp VariableDisplay
	type plain VariableDisplay
	if err := json.Unmarshal(data, (*plain)(&tmp)); err != nil {
		return err
	}
	if err := (&tmp).validate(); err != nil {
		return err
	}
	*d = tmp
	return nil
}

func (d *VariableDisplay) UnmarshalYAML(unmarshal func(interface{}) error) error {
	var tmp VariableDisplay
	type plain VariableDisplay
	if err := unmarshal((*plain)(&tmp)); err != nil {
		return err
	}
	if err := (&tmp).validate(); err != nil {
		return err
	}
	*d = tmp
	return nil
}

func (d *VariableDisplay) validate() error {
	if err := d.Display.Validate(); err != nil {
		return err
	}
	return nil
}
```

Reply:
Ok I will try this

Reply:
at the end that doesn't work just like that. Issue is tackled in #876